src/gateway.rs

```diff
@@ -207,32 +229,35 @@ impl Gateway {
             {
                 Ok(()) => info!(logger, "beacon transmitted"; "beacon" => &beacon_id),
                 Err(err) => {
+                    if let semtech_udp::server_runtime::Error::Ack(
+                        tx_ack::Error::InvalidTransmitPower(Some(value)),
+                    ) = err
+                    {
+                        if let Err(e) = self_sender.send(Message::MaxPower(value as u32)).await {
```

Comment:
use inspect_err to do the warn (it's in FutureExt iirc)

Reply:
I tried but it looks like inspect_err wants `FnOnce(&Self::Error)` which is not async. I can't handle a mutex or a channel if I'm not in an async code block.

Reply:
huh? self_sender.send is a figure right? so inspect_err would work before the await?